lib/index.js

```diff
@@ -109,6 +109,7 @@ var ReactTelephoneInput = React.createClass({
     },
     componentDidMount: function componentDidMount() {
         document.addEventListener('keydown', this.handleKeydown);
+        document.addEventListener('focus:' + this.props.name, this.onFocusEventHandler);
```

Comment:
This should happen in a wrapper component. Render the button first and listen for a click on the button at which point you hide the button, show the telephone field and set its focus.